node/src/components/network/message.rs

```diff
@@ -941,4 +941,21 @@ mod tests {
     fn bincode_roundtrip_certificate() {
         roundtrip_certificate(false)
     }
+
+    #[test]
+    fn assert_the_largest_specimen_type_and_size() {
+        let (chainspec, _) = crate::utils::Loadable::from_resources("production");
+        let estimator = NetworkMessageEstimator::new(&chainspec);
```

Comment:
This code and the lines below are almost copies of the code inside the chainspec estimation. I am a little concerned that this might make it too easy to change one, but not the other in the future. What do you think about factoring out this functionality (determining the largest specimen from a chainspec), maybe into a chainspec method or something standalone, and then using it in both places?

Reply:
I refactored the helper function so that it can be used by both.